pytests/src/datetime.rs

```diff
@@ -17,34 +17,34 @@ fn get_date_tuple<'p>(py: Python<'p>, d: &PyDate) -> Bound<'p, PyTuple> {
 }
 
 #[pyfunction]
-fn date_from_timestamp(py: Python<'_>, timestamp: i64) -> PyResult<&PyDate> {
-    PyDate::from_timestamp(py, timestamp)
+fn date_from_timestamp(py: Python<'_>, timestamp: i64) -> PyResult<Bound<'_, PyDate>> {
+    PyDate::from_timestamp_bound(py, timestamp)
 }
 
 #[pyfunction]
-fn make_time<'p>(
-    py: Python<'p>,
+fn make_time<'py>(
+    py: Python<'py>,
     hour: u8,
     minute: u8,
     second: u8,
     microsecond: u32,
-    tzinfo: Option<&PyTzInfo>,
-) -> PyResult<&'p PyTime> {
-    PyTime::new(py, hour, minute, second, microsecond, tzinfo)
+    tzinfo: Option<&Bound<'_, PyTzInfo>>,
```

Comment:
I assume the compiler is smart enough to figure out that the inner lifetime here has to be `'py`, but maybe its nicer for consistency to explicitly name it if we have it named anyway.

Reply:
Agreed, I had been tempted to leave these out as it didn't feel strictly necessary. On refection just having `'py` consistently added to all `Bound` (if the lifetime is named) is a simple rule which makes readable code 👍 

I'll update the whole file.